pkg/util/fileutil/writer_test.go

```diff
@@ -312,24 +272,153 @@ func TestWritePayloads(t *testing.T) {
 					Contents: []byte("root"),
 				},
 			},
+			second: []*v1alpha1.File{
+				{
+					Path:     "foo/bar/baz",
+					Mode:     0644,
+					Contents: []byte("second - foo"),
+				},
+				{
+					Path:     "foo/bar/2.txt",
+					Mode:     0644,
+					Contents: []byte("second - two"),
+				},
+				{
+					Path:     "foo/1.txt",
+					Mode:     0644,
+					Contents: []byte("second - one"),
+				},
+				{
+					Path:     "root.txt",
+					Mode:     0644,
+					Contents: []byte("second - root"),
+				},
+			},
+		},
+		{
+			name: "removed path - simple",
+			first: []*v1alpha1.File{
+				{
+					Path:     "foo.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			second: []*v1alpha1.File{
+				{
+					Path:     "bar.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			removed: []string{"foo.txt"},
+		},
+		{
+			name: "removed path - nested",
+			first: []*v1alpha1.File{
+				{
+					Path:     "a/foo.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			second: []*v1alpha1.File{
+				{
+					Path:     "a/bar.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			removed: []string{"a/foo.txt"},
+		},
+		{
+			name: "removed path - double nesting",
+			first: []*v1alpha1.File{
+				{
+					Path:     "a/b/foo.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			second: []*v1alpha1.File{
+				{
+					Path:     "a/c/bar.txt",
+					Mode:     0644,
+					Contents: []byte("root"),
+				},
+			},
+			removed: []string{"a/foo.txt"},
```

Comment:
`a/b/foo.txt` should not exist.

Also a test case where multiple files are deleted